tests/textarea.rs

```diff
@@ -798,3 +798,365 @@ fn test_delete_str_multiple_lines() {
+                (Key::Left, false, false, true),
+            ],
+            "い\nう",
+        ),
+    ] {
+        clear(&mut textarea);
+        textarea.insert_str(test.0);
+        for k in test.1 {
+            input_key(&mut textarea, k);
+        }
+        textarea.copy();
+        assert_eq!(textarea.yank_text(), test.2);
+    }
+}
+#[test]
+fn select_cut() {
+    let mut textarea = TextArea::default();
+    for test in [
+        // plain ascii
+        (
+            "hello world",
+            vec![
+                (Key::Home, false, false, false),
+                (Key::Right, false, false, true),
+                (Key::Right, false, false, true),
+            ],
+            "he",
+            "llo world",
+        ),
+        // plain ascii backwards
+        (
+            "hello world",
+            vec![
+                (Key::End, false, false, false),
+                (Key::Left, false, false, true),
+                (Key::Left, false, false, true),
+            ],
+            "ld",
+            "hello wor",
+        ),
+        // utf8
+        (
+            "あい",
+            vec![
+                (Key::Home, false, false, false),
+                (Key::Right, false, false, true),
+            ],
+            "あ",
+            "い",
+        ),
+        // multi line - all
+        (
+            "hello\nworld",
+            vec![
+                (Key::Char('P'), true, true, false),
+                (Key::Home, false, false, false),
+                (Key::Char('N'), true, true, true),
+                (Key::End, false, false, true),
+            ],
+            "hello\nworld",
+            "",
+        ),
+        // multi line - some
+        (
+            "hello\nworld",
+            vec![
+                (Key::Char('P'), true, true, false),
+                (Key::Home, false, false, false),
+                (Key::Right, false, false, false),
+                (Key::Char('N'), true, true, true),
+                (Key::End, false, false, true),
+                (Key::Left, false, false, true),
+            ],
+            "ello\nworl",
+            "hd",
+        ),
+        // multi - line utf8
+        (
+            "あい\nうえ",
+            vec![
+                (Key::Char('P'), true, true, false),
+                (Key::Home, false, false, false),
+                (Key::Char('N'), true, true, true),
+                (Key::End, false, false, true),
+            ],
+            "あい\nうえ",
+            "",
+        ),
+        // multi-line utf8 - some
+        (
+            "あい\nうえ",
+            vec![
+                (Key::Char('P'), true, true, false),
+                (Key::Home, false, false, false),
+                (Key::Right, false, false, false),
+                (Key::Char('N'), true, true, true),
+                (Key::End, false, false, true),
+                (Key::Left, false, false, true),
+            ],
+            "い\nう",
+            "あえ",
+        ),
+    ] {
```

Comment:
```rust
let (???, ???, ???, ???) = test;
```

Could you destructure the test case tuple at first in all tests? It is hard to understand what each elements represent without giving names to them.